src/puppeteer_crawler.js

```diff
@@ -231,13 +231,13 @@ export default class PuppeteerCrawler {
                 if (!isPromise(promise)) throw new Error('User provided handlePageFunction must return a Promise.');
 
                 return promise;
-            });
-
-        return handlePagePromise
+            })
             .timeout(this.pageOpsTimeoutMillis, 'PuppeteerCrawler: handlePageFunction timed out.')
             .finally(() => {
                 return Promise
-                    .try(() => page.close())
+                    .try(() => {
+                        if (page) return page.close();
```

Comment:
`Promise.try()` already takes care of this. It will throw on error and the `.catch()` below will handle it.

Reply:
fixed